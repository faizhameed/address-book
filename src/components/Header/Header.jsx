import React from "react";
import "./Header.scss";
const Header = ({ content }) => {
  return (
    <header>
      <h1>{content}</h1>
    </header>
  );
};

export default Header;
