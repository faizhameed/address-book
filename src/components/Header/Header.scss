:root {
  /* Base font size */
  font-size: 10px;

  /* Heading height variable*/
  --heading-height: 25em;
}

header {
  position: fixed;
  width: 100%;
  height: calc(var(--heading-height));
  z-index: 1;
  pointer-events: none;
}

/* Create angled background with 'before' pseudo-element */
header::before {
  content: "";
  display: block;
  position: absolute;
  left: 0;
  bottom: 14em;
  width: 100%;
  height: calc(var(--heading-height) + 10em);
  z-index: -1;
  transform: skewY(-3.5deg);
  background: linear-gradient(rgba(0, 0, 0, 0.6), rgba(0, 0, 0, 0.6)),
    url(https://images.unsplash.com/photo-1495464101292-552d0b52fe41?auto=format&fit=crop&w=1350&q=80)
      no-repeat center,
    linear-gradient(#4e4376, #2b5876);
  background-size: cover;
  border-bottom: 0.2em solid #fff;
  box-shadow: -1px -7px 20px 0px black;
}

h1 {
  font-size: calc(2em + 2vw);
  font-weight: 700;
  letter-spacing: 0.01em;
  padding: 20px;
  display: block;
  margin: auto;
  transform: translateY(1em);
  text-shadow: 0.022em 0.022em 0.022em #111;
  color: #fff;
  background-color: rgb(160, 71, 71);
  width: fit-content;
  border: 4px solid white;
  box-shadow: -1px -7px 20px 0px black;
}

.main {
  padding: calc(var(--heading-height)) 4em 0;
}
