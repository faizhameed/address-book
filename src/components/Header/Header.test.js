import { shallow } from "enzyme";
import Header from "./Header";
import React from "react";

describe("Header Component ", () => {
  let wrapper;
  let mockData = "Test Page for Header";
  beforeEach(() => {
    wrapper = shallow(<Header content={mockData} />);
  });

  it("should render Header component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
