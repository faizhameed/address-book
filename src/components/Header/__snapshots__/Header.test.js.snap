// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component  should render Header component 1`] = `ShallowWrapper {}`;
